lib/libvmod_blob/hex.c

```diff
@@ -99,22 +99,35 @@ hex_encode(const enum encoding enc, const enum case_e kase,
 		*p++ = alphabet[in[i] & 0x0f];
 	}
 
+	/* XXX: the length arithmetics here are a bit complicated for such
+	 * a simple problem. It looks like the estimated encoded length
+	 * makes room for a null character, which makes sense for string
+	 * encoding. But null characters are set by the caller as documented.
+	 *
+	 * The result is a confusing length computation where:
+	 *
+	 *     hex_decode_l(hex_encode_l(len)) != len
+	 *
```

Comment:
Nothing about hex or anything else in the VMOD requires this identity -- the other schemes have differing length estimates for encoding and decoding as well.

Consider especially URL, where a byte may not be changed at all, or may be encoded with three characters (%xx). The encoding length estimation makes the pessimistic assumption that *all* bytes have to be hex-encoded (nbytes*3). The decoding length estimation pessimistically assumes that *none* of them are, so that the decoding is no shorter than the encoding.